seed/models/analysis_messages.py

```diff
@@ -0,0 +1,31 @@
+from django.db import models
+
+from seed.models import (
+    Analysis,
+    AnalysisPropertyView
+)
+
+
+class AnalysisMessage(models.Model):
+    """
+    The AnalysisMessage represents user-facing messages of events that occur
+    during an analysis, like a breadcrumb trail.
+    """
+    DEFAULT = 1
+
+    MESSAGE_TYPES = (
+        (DEFAULT, 'default'),
+    )
+    # if the message is relevant to the entire analysis, ie not property specific,
+    # then it should be linked to an analysis and not linked to an AnalysisPropertyView.
+    # e.g. the API for the analysis service returned a non 200 response
+    analysis = models.ForeignKey(Analysis, on_delete=models.CASCADE)
+    # if the message is relevant to a specific property then it should be linked
+    # to an AnalysisPropertyView
+    # e.g. if the AnalysisPropertyView is missing some required data
+    analysis_property_view = models.ForeignKey(AnalysisPropertyView, on_delete=models.CASCADE, null=True, blank=True)
+    type = models.IntegerField(choices=MESSAGE_TYPES)
+    # human-readable message which is presented on the frontend
+    user_message = models.CharField(max_length=255, blank=False, default=None)
```

Comment:
I'm fine with this, but just wanted to make this note: I'm wondering if we'll ever need to have messages that are purely only meant for developers - would be easy enough to make a migration to make this to `blank=True` though.

Reply:
I think that's when we should use logging. I am even considering removing the debug_message field, but I guess we'll see if it's useful